src/components/Nav.tsx

```diff
@@ -16,6 +16,8 @@ function NavButton({
   title: string;
   label: string;
 }) {
+  const isSelected = window.location.pathname === to;
```

Comment:
I think this is built-in funtionality for NavLinks (https://reactrouter.com/en/main/components/nav-link).